### PR TITLE
ROX-14544 - Add CentralRequest field documentation

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

Added CentralRequest documentation.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Evaluated and added CHANGELOG.md entry if required
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.

## Test manual
-

### CODE REVIEW

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 1]
```diff
@@ -20,11 +20,16 @@ const (
...
+	ClusterID string `json:"cluster_id" gorm:"index"`
+	// CloudProvider ...
 	CloudProvider  string `json:"cloud_provider"`
 	CloudAccountID string `json:"cloud_account_id"`
-	MultiAZ        bool   `json:"multi_az"`
+	// MultiAZ enables multi AZ support
+	MultiAZ bool `json:"multi_az"`
+	// Name of the ACS instance
 	Name           string `json:"name" gorm:"index"`
 	Status         string `json:"status" gorm:"index"`
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:34]: Lifecycle status of the CentralRequest, link to constants

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 2]
```diff
@@ -20,11 +20,16 @@ const (
...
 	SubscriptionID string `json:"subscription_id"`
```
Comment (SimonBaeumer) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: by @kovayur 
```
SubscriptionID is returned by ams (ocm) when the user’s (owner) quota is reserved for Central
```

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 3]
```diff
@@ -20,11 +20,16 @@ const (
...
 	api.Meta
-	Region         string `json:"region"`
-	ClusterID      string `json:"cluster_id" gorm:"index"`
+	// Region is the cloud region the service is deployed in, i.e. us-east-1
+	Region string `json:"region"`
+	// ClusterID is the data-plane cluster ID
+	ClusterID string `json:"cluster_id" gorm:"index"`
+	// CloudProvider ...
 	CloudProvider  string `json:"cloud_provider"`
 	CloudAccountID string `json:"cloud_account_id"`
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:29]: Maybe it is used by @kovayur, see: 

```
rr.BillingMarketplaceAccount(dinosaur.CloudAccountID)
```

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 2 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: `Owner` = OCM owner who created the tenant (email), how is it populated? (Identify ID type, seems to be different for RH employees joined after a specific date, or based on the preferred username (OpenID feature, `ocm whoami` prints the active username))
`OwnerAccountID` and `OrganisationID` maybe the same fields. 

Code snippet:

```
dinosaurRequest.Owner, _ = claims.GetUsername()
dinosaurRequest.OrganisationID, _ = claims.GetOrgID()
dinosaurRequest.OwnerAccountID, _ = claims.GetAccountID()
dinosaurRequest.OwnerUserID, _ = claims.GetUserID()
tenantUsernameClaim = "username"

//-------------- 

// GetUsername ...
func (c *ACSClaims) GetUsername() (string, error) {
	if idx, val := arrays.FindFirst(func(x interface{}) bool { return x != nil },
		(*c)[tenantUsernameClaim], (*c)[alternateTenantUsernameClaim]); idx != -1 {
		if userName, ok := val.(string); ok {
			return userName, nil
		}
	}
	return "", fmt.Errorf("can't find neither %q or %q attribute in claims",
		tenantUsernameClaim, alternateTenantUsernameClaim)
}

//--------------

// sso.redhat.com token claim keys.
	alternateTenantUsernameClaim = "preferred_username"
	tenantUserIDClaim            = "account_id"
	tenantSubClaim               = "sub"
	// Only service accounts that have been created via the service_accounts API have this claim set.
	alternateTenantIDClaim = "rh-org-id
```

Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: `OwnerUserID`, `OwnerAccountID` needs investigation

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (stehessel) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:29]: This is the billing cloud account, definitely don't remove or we won't get paid ;-).

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 4]
```diff
@@ -20,11 +20,16 @@ const (
...
 type CentralRequest struct {
 	api.Meta
-	Region         string `json:"region"`
-	ClusterID      string `json:"cluster_id" gorm:"index"`
+	// Region is the cloud region the service is deployed in, i.e. us-east-1
+	Region string `json:"region"`
+	// ClusterID is the data-plane cluster ID
+	ClusterID string `json:"cluster_id" gorm:"index"`
+	// CloudProvider ...
 	CloudProvider  string `json:"cloud_provider"`
```
[the remark this answers is not in the document]
Comment (stehessel) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:28]: Cloud provider that runs the data plane instance. Also the cloud provider that owns the customer account we bill again.

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 2 above)
Comment (stehessel) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: organisation ID and organisation name are the Red Hat SSO organisation identifiers - it identifies a customer, e.g. `16678382` and `IBM`. We need the id for authn/z, and the name for observability purposes.

`OwnerAccountID` will be used in telemetry. `OwnerUserID` is the subject claim (confusingly it is NOT the `user_id` claim) of the Red Hat SSO token - this one I don't know if we use it anywhere.

Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: `SubscriptionID` is returned by AMS and identifies a Central instance in their system. We need it to deregister instances again from AMS.

Comment (stehessel) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: `Owner` is the Red Hat SSO login name. It is either the email, or the user name, depending on what the user chose to login with.

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:29]: done

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:28]: done

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 1 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:34]: done

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 2 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: @stehessel Where do you see the `OrganisationName`? I could find the field in the `CentralRequest` struct.

Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:35]: done

Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:39]: Rebased and added it

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 5]
```diff
@@ -20,27 +20,46 @@ const (
 // CentralRequest ...
 type CentralRequest struct {
 	api.Meta
-	Region         string `json:"region"`
-	ClusterID      string `json:"cluster_id" gorm:"index"`
-	CloudProvider  string `json:"cloud_provider"`
+	// Region is the cloud region the service is deployed in, i.e. us-east-1
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:23]: Can you end sentence comments with a "." everywhere?

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 6]
```diff
@@ -20,27 +20,46 @@ const (
...
+	// ClusterID is the data-plane cluster ID
+	ClusterID string `json:"cluster_id" gorm:"index"`
+	// CloudProvider is the cloud provider the data-plane cluster is running and is used for billing customers.
+	CloudProvider string `json:"cloud_provider"`
+	// CloudAccountID is the billing cloud account
 	CloudAccountID string `json:"cloud_account_id"`
-	MultiAZ        bool   `json:"multi_az"`
-	Name           string `json:"name" gorm:"index"`
-	Status         string `json:"status" gorm:"index"`
+	// MultiAZ enables multi AZ support
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:31]: I'd spell out availability zone in the explanation.

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 7]
```diff
@@ -20,27 +20,46 @@ const (
...
+	MultiAZ bool `json:"multi_az"`
+	// Name of the ACS instance
+	Name string `json:"name" gorm:"index"`
+	// Status is the lifecycle status of the Central request. See constants.CentralRequestStatusAccepted to see
+	// valid statuses.
+	Status string `json:"status" gorm:"index"`
+	// SubscriptionID is returned by AMS and identifies a Central instance in their system. We need it to deregister instances again from AMS.
 	SubscriptionID string `json:"subscription_id"`
-	Owner          string `json:"owner" gorm:"index"` // TODO: ocm owner?
+	// Owner is the Red Hat SSO login name. It is either the email, or the user name, depending on what the user chose to login with. It's displayed in the console UI.
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:40]:
```suggestion
	// Owner is the Red Hat SSO login name of the user who created the instance. It is either the email, or the user name, depending on what the user chose to login with. It's displayed in the console UI.
```

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 8]
```diff
@@ -20,27 +20,46 @@ const (
...
+	Owner string `json:"owner" gorm:"index"`
+	// OwnerAccountID is used in telemetry, it is the account_id claim of the Red Hat SSO token.
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:42]: We switched to `user_id` claim in telemetry. `owner_account_id` is no longer used as far as I can tell.

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 9]
```diff
@@ -20,27 +20,46 @@ const (
...
-	OwnerUserID    string `json:"owner_user_id"`
+	// OwnerUserID is the subject claim (confusingly it is NOT the user_id claim) of the Red Hat SSO token
+	OwnerUserID string `json:"owner_user_id"`
+
 	// Instance-independent part of the Central's hostname. For example, this
 	// can be `rhacs-dev.com`, `acs-stage.rhcloud.com`, etc.
-	Host             string `json:"host"`
-	OrganisationID   string `json:"organisation_id" gorm:"index"`
+	Host string `json:"host"`
+	// OrganisationID is a Red Hat SSO organisation identifier to identify a customer. It is needed as an id for authn/z, and the name for observability purposes.
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:50]:
```suggestion
	// OrganisationID is a Red Hat SSO organisation identifier to identify a customer.
```
This is not 100% true, it is just one customer organisation. For example IBM has many Red Hat orgs.

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 10]
```diff
@@ -20,27 +20,46 @@ const (
...
+	OrganisationID string `json:"organisation_id" gorm:"index"`
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:51]: This is not 100% true (depends on definition of customer), it is just one customer organisation. For example IBM has many Red Hat orgs.

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go [hunk 11]
```diff
@@ -20,27 +20,46 @@ const (
...
+	// OrganisationName is a Red Hat SSO organisation identifier to identify a customer. It is needed as an id for authn/z, and the name for observability purposes.
```
Comment (stehessel) [inline internal/dinosaur/pkg/api/dbapi/central_request_types.go:52]: Organisation names not unique. Its purpose is mostly human readability and observability purposes (e.g. display in dashboards).

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 5 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:23]: done

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 6 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:31]: done

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 7 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:40]: done

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 8 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:42]: It is never read anywhere, only set once in fleet-manager. Is it safe to assume that we can remove it?

Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:42]: How could I test that the change worked?

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 10 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:51]: Changed it to:
`// OrganisationID identifies a customer's organisation. It is needed as an id for authn/z, and the name for observability purposes.`

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 8 above)
Comment (stehessel) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:42]: > How could I test that the change worked?

You can take a look at Telemetry events in https://app.segment.com/redhat-devtools/sources/acs_instances_backend_dev/debugger.

> Is it safe to assume that we can remove it?

I think we can remove this field, yes. Just need to be careful with backwards/forwards compatibility.

**Review by stehessel — APPROVED** [review]
Thanks for going over these 🚀

File: internal/dinosaur/pkg/api/dbapi/central_request_types.go (hunk 8 above)
Comment (stehessel) [reply internal/dinosaur/pkg/api/dbapi/central_request_types.go:42]: Yeah that makes sense.

**Pull request merged**